src/Analyser/TypeSpecifier.php

```diff
@@ -818,6 +818,60 @@ public function specifyTypesInCondition(
 								$rootExpr,
 							),
 						);
+					} elseif ($var->dim instanceof Expr\Variable) {
```

Comment:
Doesn't have to be just a variable. Getting of `instanceof` like these leads to bugfixes:

* https://github.com/phpstan/phpstan-src/commit/d3a2a92fcd612bf42bbfd19cd3a5625481ff7522?w=1
* https://github.com/phpstan/phpstan-src/commit/00d2caf39514380610899b0a305413f60c1c5830?w=1 (regression tests in a follow-up commit)

Reply:
https://github.com/phpstan/phpstan/issues/5734 should be fixed then too